File: algorithms/insertion_sort/insertion_sort.go
package main

import "fmt"

func InsertionSort(arr []int) []int {

	for i := 1; i < len(arr); i++ {
		tempValue := arr[i]
		comparisonPos := i - 1

		for comparisonPos >= 0 {
			if arr[comparisonPos] > tempValue {
				arr[comparisonPos+1] = arr[comparisonPos]
				comparisonPos--
			} else {
				break
			}
		}
		arr[comparisonPos+1] = tempValue
	}
	return arr
}

func main() {
	fmt.Println(InsertionSort([]int{4, 2, 7, 1, 3}))
}

File: algorithms/insertion_sort/insertion_sort.py
def insertion_sort(arr):
    # Iterate over the array
    for i in range(1, len(arr)):
        temp = arr[i]
        position = i - 1
        
        # Iterate backwards from the position of the pointer
        while position >= 0:
            if arr[position] > temp:
                arr[position + 1] = arr[position]
                position =- 1
            else:
                break
        arr[position + 1] = temp
